src/styles/main.scss

```diff
@@ -0,0 +1,62 @@
+$day-size: 100px;
+$border-color: #000;
+$default-bg-color: #eee;
+$hover-bg-color: #ffbfcb;
+$font-size: 30px;
+$font-family: Arial, sans-serif;
+$columns: 7;
+$padding: 10px;
+$animation-duration: 0.5s;
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: calc(#{$columns} * (#{$day-size} + 1px) - 1px);
+  margin: 0 auto;
+  align-items: center;
+  counter-reset: day 0;
+
+  &__day {
+    width: $day-size;
+    height: $day-size;
+    background-color: $default-bg-color;
+    border: 1px solid $border-color;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    font-family: $font-family;
+    font-size: $font-size;
+    position: relative;
+    cursor: pointer;
+    transition:
+      transform $animation-duration,
+      background-color $animation-duration;
+    counter-increment: day;
+
+    &::before {
+      content: counter(day);
+    }
+
+    &:hover {
+      background-color: $hover-bg-color;
+      transform: translateY(-20px);
+    }
+  }
+
+  // Позиціюємо перший день у залежності від того, з якого дня починається місяць
+  @each $day, $index in (mon: 0, tue: 1, wed: 2, thu: 3, fri: 4, sat: 5, sun: 6)
```

Comment:
The `@each` loop is used to dynamically create classes for different starting days of the week. Ensure that these classes are correctly applied in the HTML to achieve the desired layout.